### PR TITLE
Add metrics for locks

### PR DESCRIPTION
**Pull request by topherbullock**
More annoying to add than expected, since the metrics package imports the db package (for good reason), so needed to inject a "`LogFunc`" into the `LockFactory`.

### CODE REVIEW

**Comment by topherbullock** [comment]
Summoning @TimSimmons and @databus23 for their opinions on using a `GaugeVec` for Prometheus  metrics, since you're both more knowledgable than I am about Prometheus. 😄

File: atc/metric/metrics.go [hunk 1]
```diff
@@ -415,3 +416,79 @@ func (event ResourceCheck) Emit(logger lager.Logger) {
...
+			Attributes: map[string]string{
+				"type":      event.LockType,
+				"object_id": strconv.Itoa(event.ObjectID),
+			},
+		},
+	)
+}
+
+func LogLockAcquired(logger lager.Logger, lockID lock.LockID) {
+	logger.Debug("released")
```
Comment (cirocosta) [inline atc/metric/metrics.go]: whoops? "acquired" maybe 😁

**Comment by cirocosta** [comment]
Nice!

Something that crossed my mind: would it be interesting to metrify the act of not being able to acquire a lock?

In the sense that we'd be able to quickly spot that a particular type of lock is being requested but never being acquired.

wdyt?

thx!

File: atc/metric/metrics.go (hunk 1 above)
Comment (topherbullock) [reply atc/metric/metrics.go]: oops

**Comment by topherbullock** [comment]
> (I went ahead and resolved the conflicts to kick the build again.)
> 
@vito 
THANKS, BOO

**Comment by jama-pivotal** [comment]
linking to #2674

**Pull request merged**